py/desispec/quicklook/qlconfig.py

```diff
@@ -244,95 +268,143 @@ def io_qa(self,qaname):
+
+        outconfig['PipeLine'] = pipeline
+        outconfig['RawImage'] = self.rawfile
+        outconfig["OutputFile"] = self.outputfile
+        outconfig['Timeout'] = self.timeout
+
+        #- Check if all the files exist for this QL configuraion
+        check_config(outconfig)
+
+        return outconfig
+
+def check_config(outconfig):
+    """
+    Given the expanded config, check for all possible file existence etc....
+    """
+
+    if outconfig["Flavor"]=="dark":
+        files = [outconfig["RawImage"], outconfig["FiberMap"], outconfig["FiberFlatFile"], outconfig["PSFFile"]]
+        log.info("Checking if all the necessary files exist.")
+        for thisfile in files:
+            if not os.path.exists(thisfile):
+                sys.exit("File does not exist: {}".format(thisfile))
+            else:
+                log.info("File check: Okay: {}".format(thisfile))
+        log.info("All necessary file exist for this configuration.")
+    return 
+
+
 class Palist(object):
     
     """
     Generate PA list and QA list for the Quicklook Pipeline for the given exposure
     """
-    def __init__(self,flavor,mode="online"):
+    def __init__(self,thislist=None,algorithms=None,flavor=None,mode=None):
         """
-        flavor: flavor can be arc,flat, dark, lrg, elg etc....
-        mode: mode can be one in ["online", "offline"]
+        thislist: given list of PAs
+        algorithms: Algorithm list coming from config file: e.g desispec/data/quicklook/quicklook.darktime.yaml
+        flavor: only needed if new list is to be built.
+        mode: online offline?
         """
         self.flavor=flavor
         self.mode=mode
+        self.thislist=thislist
+        self.algorithms=algorithms
         self.palist=self._palist()
         self.qalist=self._qalist()
-
+        
     def _palist(self):
-
-        if self.flavor == "arcs":
-            pa_list=['Initialize','Preproc','BoxcarExtract'] #- class names for respective PAs (see desispec.quicklook.procalgs)
-        elif self.flavor == "flat":
-            pa_list=['Initialize','Preproc','BoxcarExtract', 'ComputeFiberFlat_QL']
-        elif self.flavor in ['dark','elg','lrg','qso','bright','grey','gray','bgs','mws']:
-            pa_list=['Initialize','Preproc','BoxcarExtract', 'ApplyFiberFlat_QL','SkySub_QL']
-        else:
-            log.warning("Not a valid flavor type. Use a valid flavor type. Exiting.")
-            sys.exit(0)
+        
+        if self.thislist is not None:
+            pa_list=self.thislist
+        else: #- construct palist
+            if self.flavor == "arcs":
+                pa_list=['Initialize','Preproc','BoxcarExtract'] #- class names for respective PAs (see desispec.quicklook.procalgs)
+            elif self.flavor == "flat":
+                pa_list=['Initialize','Preproc','BoxcarExtract', 'ComputeFiberFlat_QL']
+            elif self.flavor in ['dark','elg','lrg','qso','bright','grey','gray','bgs','mws']:
```

Comment:
@profxj  Yes, we realized this change was coming.  We need to keep track of the difference between dark, bright and grey exposures (and also arc, continuum lamp, etc) because each pipeline will be configured differently.  The science pipelines will all likely run the same algorithms, but it is likely the params will be different (eg. targets to focus on will certainly change), so we need a term to distinguish dark, bright and grey.  We will stop using 'flavor' for this and instead use 'obstype' for observation type unless the recent changes in #386 also provide a new term for this (this was not obvious to me).